File: python/leetcode/add-digits.py
class Solution:
    def addDigits(self, num: int) -> int:
        result = num
        sum = num
        while num > 9:
            iteration_num = num
            sum = 0
            while iteration_num > 0:
                r = iteration_num % 10
                sum += r
                iteration_num = (iteration_num - r) // 10
            num = sum
        return sum

File: java/leetcode/add-digits.java
class Solution {
    public int addDigits(int num) {
        int result = num;
        int sum = num;
        while (num > 9) {
            int iterationNum = num;
            sum = 0;
            while (iterationNum > 0) {
                int r = iterationNum % 10;
                sum += r;
                iterationNum = (iterationNum - r) / 10;            
            }
            num = sum;
        }
        return sum;
    }
}

